File: src/main/java/com/aoranzhang/ezrentback/data/entity/City.java
package com.aoranzhang.ezrentback.data.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import java.util.Set;

@Entity
@Table(name = "city")
public class City extends AbstractPersistentObject{
    private String name;

    private Double longitude, latitude;

    @OneToMany(
            mappedBy = "city",
            cascade = CascadeType.ALL,
            orphanRemoval = true,
            fetch = FetchType.LAZY
    )
    @JsonIgnore
    private Set<Building> buildings;

    public City() {}

    public City(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Double getLongitude() {
        return longitude;
    }

    public void setLongitude(Double longitude) {
        this.longitude = longitude;
    }

    public Double getLatitude() {
        return latitude;
    }

    public void setLatitude(Double latitude) {
        this.latitude = latitude;
    }

    public Set<Building> getBuildings() {
        return buildings;
    }

    public void setBuildings(Set<Building> buildings) {
        this.buildings = buildings;
    }
}

File: src/main/kotlin/com/aoranzhang/ezrentback/data/entity/City.kt
package com.aoranzhang.ezrentback.data.entity

import com.fasterxml.jackson.annotation.JsonIgnore

import javax.persistence.*

@Entity
@Table(name = "city")
class City : AbstractPersistentObject {
    var name: String = ""

    var longitude: Double = 0.0
    var latitude: Double = 0.0

    @OneToMany(mappedBy = "city", cascade = [CascadeType.ALL], orphanRemoval = true, fetch = FetchType.EAGER)
    @JsonIgnore
    var buildings: Set<Building> = HashSet()


    constructor() : super() {}

    constructor(city: City) : super(city.id, city.version) {
        this.name = city.name
        this.longitude = city.longitude
        this.latitude = city.latitude
        this.buildings = city.buildings
    }

    constructor(name: String, longitude: Double, latitude: Double, buildings: Set<Building>) : super() {
        this.name = name
        this.longitude = longitude
        this.latitude = latitude
        this.buildings = buildings
    }

}